src/Editor/Editor.js

```diff
@@ -187,16 +176,20 @@ function Editor({
   // onValidate
   useEffect(() => {
     if (isEditorReady) {
-      monacoRef.current.editor.setModelMarkers = function(model, owner, markers) {
-        getModelMarkersSetter().backup?.call(
-          monacoRef.current.editor,
-          model,
-          owner,
-          markers,
-        );
-
-        onValidate?.(markers);
-      }
+      const changeMarkersListener = monacoRef.current.editor.onDidChangeMarkers(uris => {
+        const editorUri = editorRef.current?.getModel()?.uri;
+        if (editorUri) {
```

Comment:
I have questions about these two checks:

1) line 180 - `= editorRef.current?.getModel()?.uri;`
2) line 181 - `if (editorUri) {`

<hr />

From the first glance `editorRef.current?` check is unnecessary because in line 178 we have `if (isEditorReady) {`, which ensures the existence of `monacoRef.current`. Based on the previous statement it turns out that `getModel()?` and `if (editorUri) {` are also unnecessary. But theoretically, it's possible to dispose `editor` and all its related models from the outside (as users have access to both `monaco` and `editor` instances).

I would like to know your thoughts about these checks and know what were your motives when you add them? 🙂

Reply:
Hey, sure. You are correct I do not need to add the check for `editorRef.current` as that is already checked in the `isEditorReady` that is a mistake by me, sorry. 
Also, the typing on the `getModel` method is `ITextModel | null`. I was just being safe even though if we know we have `editorRef.current` you would think we should always be able to get its model.

Reply:
okay, in that case, I'll remove the question mark after `editorRef.current` and will leave the rest as is.